garage/envs/base.py

```diff
@@ -1,17 +1,132 @@
+"""Wrapper class that converts gym.Env into GarageEnv."""
 import collections
 
-from cached_property import cached_property
+import gym
+from gym.spaces import Box as GymBox
+from gym.spaces import Discrete as GymDiscrete
+from gym.spaces import Tuple as GymTuple
+
+from garage.core import Parameterized
+from garage.core import Serializable
+from garage.spaces import Box
+from garage.spaces import Discrete
+from garage.spaces import Product
+
+
+class GarageEnv(gym.Wrapper, Parameterized, Serializable):
+    """
+    Returns a Garage wrapper class for gym.Env.
+
+    In order to provide pickling (serialization) and parameterization
+    for gym.Envs, they must be wrapped with GarageEnv. This ensures
+    compatibility with existing samplers and checkpointing when the
+    envs are passed internally around garage.
+
+    Furthermore, GarageEnv silently converts action_space and
+    observation_space from gym.Spaces to garage.spaces.
+
+    Args:
+        env (gym.Env): the env that will be wrapped
+    """
+
+    def __init__(self, env):
+        super(GarageEnv, self).__init__(env)
+        Parameterized.__init__(self)
+        Serializable.quick_init(self, locals())
+        self._wrapped_env = env
+        self.action_space = self.env.action_space
+        self.observation_space = self.env.observation_space
+        self.reward_range = self.env.reward_range
+        self.metadata = self.env.metadata
+
+    def close(self):
+        """
+        Close the wrapped env.
+
+        Returns:
+            None
+        """
+        self._wrapped_env.close()
+
+    def get_params_internal(self, **tags):
+        """
+        Returns an empty list if env.get_params() is called.
+
+        Returns:
+            An empty list
+        """
+        return []
+
+    @property
+    def horizon(self):
+        """
+        Get the maximum episode steps for the wrapped env.
+
+        Returns:
+            max_episode_steps (int)
+        """
+        return self._wrapped_env.spec.tags[
+            'wrapper_config.TimeLimit.max_episode_steps']
+
+    def log_diagnostics(self, paths, *args, **kwargs):
```

Comment:
log_diagnostics() is not part of the gym interface so you can't call it on the underlying env. 

If this is necessary because of calls elsewhere, implement it as pass and then make an issue to remove all said calls.